File: 백준/Silver/1193. 분수찾기/분수찾기.py
# 1/1
# 1/2, 2/1
# 3/1, 2/2, 1/3
# 1/4, 2/3, 3/2, 4/1
# 5/1, 4/2, 3/3, 2/4, 1/5

# 줄위치 신경쓰자, N > line 루프가 풀리면 if ~ else는 while에 종속되지 않고 작동


N = int(input())

def converter(N):
    line = 1
    
    while N > line:
        N -= line
        line += 1

    if line % 2 == 0:
        top = N
        bottom = line - N + 1

    else:
        top = line - N + 1
        bottom = N

    return top, bottom


top, bottom = converter(N)
print(f"{top}/{bottom}")

File: 백준/Silver/1193. 분수찾기/분수찾기.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int X = Integer.parseInt(br.readLine());
		String res = fraction(X);
		System.out.println(res);
	}
	
	public static String fraction(int X) {
		int line = 1;
		String answer = "";
		int top = 0;
		int bottom = 0;
		
		
		while (X > line) {
			X -= line;
			line++;
		}
		
		if (line % 2 == 1) {
			top = line - X + 1;
			bottom = X;
		}
		else if (line % 2 == 0) {
			top = X;
			bottom = line - X + 1;
		}
		
		answer = top + "/" + bottom;
		
		return answer;
	}
	
	
}